RedHatInsights/insights-inventory-frontend
# Feat(RHINENG-4796): Add Public IP & FQDN to Infrastructure Card in Host Details Page

# Description

Add Public IP & FQDN to Infrastructure Card in Host Details Page
The necessary changes that address the issue described on the ticket

Associated Jira ticket: [#RHINENG-4796](https://issues.redhat.com/browse/RHINENG-4796) 

# How to test the PR

Ensure a host containing the properties `public_ipv4_addresses` (& `public_dns`) is presenting those properties on its Infrastructure Card in its Host Details page
(It could also be with two different hosts that each contain one property)

Expected Functionality: 
- Hosts that have a property, should present it.
- Hosts that do not have a property, should present 'Not available' instead.

To test, you can search for a host named `lior_test_ip-172-31-89-28.ec2.internal` on `insights-qa` account. 
This host contains the necessary `public_ipv4_addresses` property that should be presented on the Infrastructure Card.

### Please Read
*However, because we are not yet collecting the `public_dns`, we can not create a host that contain this property.
Because both of these properties are part of the same SystemProfile object that is received in the relevant API response, both properties have the same implementation in this PR, which should work.*

*Here is a link to the scheme of `SystemProfile` that is being returned:
https://github.com/lzap/inventory-schemas/blob/fe94ba7ab5561abf1e1d644645b0110d6e64fbc0/schemas/system_profile/v1.yaml#L389-L406*

*The API endpoint that receives SystemProfile that SHOULD contain both properties, but now containing only 1 because this host does not contain the `public_dns` property (see the screenshots attached):*

![Screenshot from 2024-01-08 15-18-50](https://github.com/RedHatInsights/insights-inventory-frontend/assets/93318917/e93fb9b5-7cfc-44ce-977b-ab5602c4295a)

![Screenshot from 2024-01-08 15-19-06](https://github.com/RedHatInsights/insights-inventory-frontend/assets/93318917/ff0c03f1-3f0f-425f-950f-d45037bfda2f)


*Tested locally that when a host has also the `public_dns` property, it is being presented in the UI alongside the `public_ipv4_addresses` property. 
Attached screenshots of the changes and the UI:*

*Added code that is needed to ensure that the necessary values are presented when received*

![Added code that is needed to ensure that the necessary values are presented when received](https://github.com/RedHatInsights/insights-inventory-frontend/assets/93318917/a4371863-aa0c-4ca5-afd8-4c22a18f2413)

*The passed values are presented in the UI*

![The passed values are presented in the UI](https://github.com/RedHatInsights/insights-inventory-frontend/assets/93318917/6a81baee-26e7-4008-8aae-61fe09520c05)

# Before the change

Host Details Page for a host that contains the `public_ipv4_addresses` property
Green - name of host so we can easily find it
Yellow - Infrastructure Card does not contain the mentioned properties

![Before: Host Details Page for a host that contains the public_ipv4_addresses property](https://github.com/RedHatInsights/insights-inventory-frontend/assets/93318917/fb672923-57c9-49f6-9c5a-2dd936cbcc84)

# After the change

Host Details Page for a host that contains the `public_ipv4_addresses` property
Green - name of host so we can easily find it (same host name)
Orange - New Public IP property that is showing on the Infrastructure Card because this host contains the necessary property
Yellow - New FQDN property is being presented on the card, but is showing Not Available for this host because it does not contain the necessary `public_dns` property

![After: Host Details Page for a host that contains the public_ipv4_addresses property](https://github.com/RedHatInsights/insights-inventory-frontend/assets/93318917/68765b8c-4845-4d1d-8082-5d8566ea13f9)


# Checklist

- [X] Check the property `public_ipv4_addresses`
- [X] Check the property `public_dns` -> Need to create a host containing this prop (blocked) but tested locally (explained in the Please Read section)
- [X] update appropriate snapshots
- [X] Add screenshots
